src/tests/D3MHub.t.sol

```diff
@@ -1293,315 +1144,28 @@ contract D3MHubTest is DssTest {
-        assertGt(opink, 0);
-        assertGt(opart, 0);
-
-        (uint256 npink, uint256 npart) = vat.urns(ilk, address(newPool));
-        assertEq(npink, 0);
-        assertEq(npart, 0);
-        assertTrue(newPool.preDebt() == false);
-        assertTrue(newPool.postDebt() == false);
-
-        // quit to new pool
-        d3mTestPool.quit(address(newPool));
-        vat.grab(
-            ilk,
-            address(d3mTestPool),
-            address(newPool),
-            address(newPool),
-            -int256(opink),
-            -int256(opart)
-        );
-        vat.grab(
-            ilk,
-            address(newPool),
-            address(newPool),
-            address(newPool),
-            int256(opink),
-            int256(opart)
-        );
-
-        // Ensure we quit our position
-        (opink, opart) = vat.urns(ilk, address(d3mTestPool));
-        assertEq(opink, 0);
-        assertEq(opart, 0);
-        // quit does not call hooks
-        assertTrue(d3mTestPool.preDebt() == false);
-        assertTrue(d3mTestPool.postDebt() == false);
-
-        (npink, npart) = vat.urns(ilk, address(newPool));
-        assertEq(npink, 50 * WAD);
-        assertEq(npart, 50 * WAD);
-        assertTrue(newPool.preDebt() == false);
-        assertTrue(newPool.postDebt() == false);
-
-        // file new pool
-        d3mHub.file(ilk, "pool", address(newPool));
-
-        // test unwind/wind
-        d3mTestPlan.file("targetAssets", 45 * WAD);
-        d3mHub.exec(ilk);
-
-        (opink, opart) = vat.urns(ilk, address(d3mTestPool));
-        assertEq(opink, 0);
-        assertEq(opart, 0);
-
-        (npink, npart) = vat.urns(ilk, address(newPool));
-        assertEq(npink, 45 * WAD);
-        assertEq(npart, 45 * WAD);
-
-        d3mTestPlan.file("targetAssets", 100 * WAD);
-        d3mHub.exec(ilk);
-
-        (opink, opart) = vat.urns(ilk, address(d3mTestPool));
-        assertEq(opink, 0);
-        assertEq(opart, 0);
-
-        (npink, npart) = vat.urns(ilk, address(newPool));
-        assertEq(npink, 100 * WAD);
-        assertEq(npart, 100 * WAD);
-    }
-
-    function test_plan_upgrade() public {
-        _windSystem(); // Tests that the current pool has ink/art
-
-        // Setup new plan
-        D3MTestPlan newPlan = new D3MTestPlan(address(dai));
-        newPlan.file("maxBar_", type(uint256).max);
-        newPlan.file("bar", 5);
-        newPlan.file("targetAssets", 100 * WAD);
-
-        d3mHub.file(ilk, "plan", address(newPlan));
-
-        (, ID3MPlan plan, , , ) = d3mHub.ilks(ilk);
-        assertEq(address(plan), address(newPlan));
-
-        d3mHub.exec(ilk);
-
-        // New Plan should determine the pool position
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(d3mTestPool));
-        assertEq(ink, 100 * WAD);
-        assertEq(art, 100 * WAD);
-        assertTrue(d3mTestPool.preDebt());
-        assertTrue(d3mTestPool.postDebt());
-    }
-
-    function test_hub_upgrade_same_d3ms() public {
```

Comment:
Is there a problem to keep this one?

Reply:
Same as above, I think this should be an integration test. This as a unit test provides a false sense of coverage imo, but if you really want to readd I won't die on this hill.

Reply:
Let's wait for @rockyfour comment.

Reply:
Sure. I will be adding this integration test in a future PR btw as Hub needs to have minor upgrades to support the Star Vaults.

Reply:
See my comment in the above discussion about `test_quit_culled`.

Reply:
Resolving as it was re-added.